### PR TITLE
Some refactoring

### PR DESCRIPTION
**Pull request by simogasp**
* optimized sampling function(s)
* using eigen `hnormalize()` and `homogeneous()`
* some random cleaning / formatting

### CODE REVIEW

File: src/openMVG/multiview/projection.cpp [hunk 1]
```diff
@@ -25,29 +25,44 @@
...
 
   // Set K(2,1) to zero.
-  if (K(2,1) != 0) {
+  if (K(2,1) != 0)
+  {
     double c = -K(2,2);
     double s = K(2,1);
-    double l = sqrt(c * c + s * s);
-    c /= l; s /= l;
+    const double lenght = std::sqrt(c * c + s * s);
```
Comment (fabiencastan) [inline src/openMVG/multiview/projection.cpp]: length

File: src/openMVG/multiview/rotation_averaging_test.cpp [hunk 2]
```diff
@@ -104,13 +104,13 @@ TEST ( rotation_averaging, RefineRotationsAvgL1IRLS_SimpleTriplet)
   using namespace std;
 
   //--
-  // Setup 3 camera that have a relative orientation of 120�
+  // Setup 3 camera that have a relative orientation of 120�
```
Comment (fabiencastan) [inline src/openMVG/multiview/rotation_averaging_test.cpp]: deg

File: src/openMVG/multiview/rotation_averaging_test.cpp [hunk 3]
```diff
@@ -104,13 +104,13 @@ TEST ( rotation_averaging, RefineRotationsAvgL1IRLS_SimpleTriplet)
...
   // Set Z axis as UP Vector for the rotation
   // They are in the same plane and looking in O={0,0,0}
   //--
-  Mat3 R01 = RotationAroundZ(2.*M_PI/3.0); //120�
-  Mat3 R12 = RotationAroundZ(2.*M_PI/3.0); //120�
-  Mat3 R20 = RotationAroundZ(2.*M_PI/3.0); //120�
+  Mat3 R01 = RotationAroundZ(2.*M_PI/3.0); //120�
```
Comment (fabiencastan) [inline src/openMVG/multiview/rotation_averaging_test.cpp]: deg

File: src/openMVG/multiview/triangulation_nview.cpp [hunk 4]
```diff
@@ -30,120 +30,127 @@
...
+  {
+    const Mat34& PMat = views[i].first;
+    const Vec2 & xy = views[i].second;
+    const Vec2 p = Project(PMat, X);
+    squared_reproj_error += (xy - p).norm();
+  }
+  return squared_reproj_error;
+}
+
+// Camera triangulation using the iterated linear method
```
Comment (fabiencastan) [inline src/openMVG/multiview/triangulation_nview.cpp]: documentation on functions should use doxygen formatting `///` or `/**` and should be done in header.

File: src/openMVG/robust_estimation/rand_sampling.hpp [hunk 5]
```diff
@@ -25,85 +25,156 @@
...
+    std::unordered_set<IntT> samples;
+    for(IntT d = upperBound - numSamples; d < upperBound; ++d)
+    {
+      IntT t = std::uniform_int_distribution<>(0, d)(generator) + lowerBound;
+      if(samples.find(t) == samples.end())
+        samples.insert(t);
+      else
+        samples.insert(d);
+    }
+    assert(samples.size() == numSamples);
```
Comment (fabiencastan) [inline src/openMVG/robust_estimation/rand_sampling.hpp]: You add `numSamples` features in a container, then you remove duplicates.
So `samples.size() <= numSamples`.

File: src/openMVG/robust_estimation/rand_sampling.hpp [hunk 6]
```diff
@@ -25,85 +25,156 @@
...
   {
-    IntT sample = uniform_dist(e1);
-    if(set_samples.count(sample) == 0)
-    {
-      set_samples.insert(sample);
-      samples->push_back(sample);
-    }
+    samples.insert(s);
   }
+  assert(samples.size() == numSamples);
```
Comment (fabiencastan) [inline src/openMVG/robust_estimation/rand_sampling.hpp]: samples is an std::set, so `samples.size() <= numSamples`.

File: src/openMVG/robust_estimation/robust_estimator_ACRansac_test.cpp [hunk 7]
```diff
@@ -214,17 +216,16 @@ TEST(RansacLineFitter, RealisticCase)
   }
 
   // Setup a normal distribution in order to make outlier not aligned
-  std::random_device rd;
-  std::mt19937 gen(rd());
+  std::mt19937 gen;
   std::normal_distribution<> d(0, 5); // More or less 5 units
 
   //-- Simulate outliers (for the asked percentage amount of the datum)
   const int nbPtToNoise = (int) NbPoints * inlierPourcentAmount / 100.0;
```
Comment (fabiencastan) [inline src/openMVG/robust_estimation/robust_estimator_ACRansac_test.cpp]: could use a ratio directly instead of inlierPourcentAmount.

File: src/openMVG/robust_estimation/robust_estimator_MaxConsensus_test.cpp [hunk 8]
```diff
@@ -91,22 +91,22 @@ TEST(MaxConsensusLineFitter, TooFewPoints) {
 //  Check that the number of inliers and the model are correct.
 TEST(MaxConsensusLineFitter, RealisticCase) {
 
-  const int NbPoints = 30;
-  const int inlierPourcentAmount = 30; //works with 40
-  Mat2X xy(2, NbPoints);
+  const int numPoints = 30;
+  const int outlierRatio = 30; //works with 40
```
Comment (fabiencastan) [inline src/openMVG/robust_estimation/robust_estimator_MaxConsensus_test.cpp]: const float outlierRatio = 0.3;

File: src/openMVG/robust_estimation/rand_sampling.hpp (hunk 5 above)
Comment (simogasp) [inline src/openMVG/robust_estimation/rand_sampling.hpp]: no it must be `==`. It must always draw exactly numSamples element.
in the `for` it added numSample elements that are guaranteed to be unique (see if...else)

File: src/openMVG/robust_estimation/rand_sampling.hpp (hunk 6 above)
Comment (simogasp) [inline src/openMVG/robust_estimation/rand_sampling.hpp]: same as above, it is `==`, this is a wrapper around the same function in order to get a `std::set` as output instead of a `std::vector`, because somewhere in the lib there is a case in which the samples must be ordered. It could be also inside the if
```
assert(samples.insert(s).second)
```
as the vector vecSamples must have only unique elements

File: src/openMVG/robust_estimation/rand_sampling.hpp (hunk 5 above)
Comment (simogasp) [inline src/openMVG/robust_estimation/rand_sampling.hpp]: but actually a good point is that i can use directly a `std::vector` instead of a `std::unordered_set` and spare the final copy.

File: src/openMVG/multiview/projection.cpp (hunk 1 above)
Comment (zvrba) [inline src/openMVG/multiview/projection.cpp]: Using this http://en.cppreference.com/w/cpp/numeric/math/hypot would be even better because it takes care of avoiding overflow/underflow in the intermediate stages.

Comment (simogasp) [inline src/openMVG/multiview/projection.cpp]: thanks, I didn't know this function (like many others :-D)

File: src/openMVG/multiview/triangulation_nview.cpp (hunk 4 above)
Comment (simogasp) [inline src/openMVG/multiview/triangulation_nview.cpp]: i made those changes/doc in the IRLS triangulation, gonna be another PR
